adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/JDBCSpace.java

```diff
@@ -48,208 +43,170 @@ public class JDBCSpace implements AutoCloseable {
+            connConfig.addDataSourceProperty("sslmode", "prefer");
         }
 
         Optional<String> sslCert = cfg.getOptional("sslcert");
         if (sslCert.isPresent()) {
-            hikariConfig.addDataSourceProperty("sslcert", sslCert.get());
+            connConfig.addDataSourceProperty("sslcert", sslCert.get());
         } /*else if(sslMode.isPresent() && (!"disable".equalsIgnoreCase(sslMode.get()) || !"allow".equalsIgnoreCase(sslMode.get())) || !"prefer".equalsIgnoreCase(sslMode.get())) {
             throw new OpConfigError("When sslmode is true, sslcert should be provided.");
         }*/
 
         Optional<String> sslRootCert = cfg.getOptional("sslrootcert");
         if (sslRootCert.isPresent()) {
-            hikariConfig.addDataSourceProperty("sslrootcert", sslRootCert.get());
+            connConfig.addDataSourceProperty("sslrootcert", sslRootCert.get());
         }
 
-        hikariConfig.addDataSourceProperty("applicationName", cfg.get("applicationName"));
-        hikariConfig.addDataSourceProperty("rewriteBatchedInserts", cfg.getOrDefault("rewriteBatchedInserts", true));
+        connConfig.addDataSourceProperty("applicationName", cfg.get("applicationName"));
+        connConfig.addDataSourceProperty("rewriteBatchedInserts", cfg.getOrDefault("rewriteBatchedInserts", true));
 
-        // We're managing the auto-commit behavior of connections ourselves and hence disabling the auto-commit.
-        Optional<String> autoCommitOpt = cfg.getOptional("autoCommit");
-        boolean autoCommit = false;
-        if (autoCommitOpt.isPresent()) autoCommit = BooleanUtils.toBoolean(autoCommitOpt.get());
-        hikariConfig.setAutoCommit(autoCommit);
-
-        hikariConfig.setKeepaliveTime(Integer.parseInt(cfg.get("keepaliveTime")));
-
-        // HikariCP "maximumPoolSize" parameter is ignored.
+        connConfig.setKeepaliveTime(Integer.parseInt(cfg.get("keepaliveTime")));
         // Use the NB "num_conn" parameter instead, wth 20% extra capacity
-        hikariConfig.setMaximumPoolSize((int)Math.ceil(1.2*maxNumConn));
+        connConfig.setMaximumPoolSize((int) Math.ceil(1.2 * maxNumConn));
 
-        this.hikariDataSource = new HikariDataSource(hikariConfig);
+        if (useHikariCP) {
+            this.hikariDataSource = new HikariDataSource(connConfig);
+            logger.info("hikariDataSource is created : {}", hikariDataSource);
+        }
     }
 
     private void shutdownSpace() {
-        isShuttingDown = true;
-
         try {
-            waitUntilAllOpFinished(System.currentTimeMillis());
-
+            logger.info("Shutting down JDBCSpace -- total {} of connections is being closed ...", connections.size());
             for (Connection connection : connections.values()) {
-                connection.close();
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Close connection : {}", connection);
+                    connection.close();
```

Comment:
This closing logic will not be applied if the logger doesn't have debug enabled

Reply:
Thanks. Fixed